frontend/ui/screensaver.lua

```diff
@@ -450,6 +450,33 @@ function Screensaver:setup(event, event_message)
                 doc_settings = DocSettings:open(lastfile)
             end
             excluded = doc_settings:isTrue("exclude_screensaver")
+
+            local book_summary = doc_settings:readSetting("summary")
+            local book_finished = book_summary and book_summary.status == "complete"
+            local book_on_hold = book_summary and book_summary.status == "abandoned"
+            if not excluded then
+                if not ui and G_reader_settings:isTrue("screensaver_hide_cover_in_filemanager") then
+                    excluded = true
+                elseif G_reader_settings:isTrue("screensaver_exclude_finished_books") and book_finished then
+                    excluded = true
+                elseif G_reader_settings:isTrue("screensaver_exclude_on_hold_books") and book_on_hold then
+                    excluded = true
+                end
+                if excluded then
+                    self.show_message = false
+                end
+            else
+                -- doc_settings:isTrue("exclude_screensaver") does not get rid of the message, so honor it, when not in any of the other cases
+                if self.show_message then
+                    if not ui and G_reader_settings:isTrue("screensaver_hide_cover_in_filemanager") then
+                        self.show_message = false
+                    elseif G_reader_settings:isTrue("screensaver_exclude_finished_books") and book_finished then
+                        self.show_message = false
+                    elseif G_reader_settings:isTrue("screensaver_exclude_on_hold_books") and book_on_hold then
+                        self.show_message = false
+                    end
+                end
+            end -- if not excluded
```

Comment:
Looks like these 2 big similar branches can be merged ? (more checks even if already excluded, but it avoids duplicating the whole thing):
```lua
            local book_on_hold = book_summary and book_summary.status == "abandoned"
            if not ui and G_reader_settings:isTrue("screensaver_hide_cover_in_filemanager") then
                excluded = true
            elseif G_reader_settings:isTrue("screensaver_exclude_finished_books") and book_finished then
                excluded = true
            elseif G_reader_settings:isTrue("screensaver_exclude_on_hold_books") and book_on_hold then
                excluded = true
            end
            if excluded then
                self.show_message = false
            end
```